Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -0,0 +1,446 @@
+
+        QLabel* thirdRightPaneHeader = new QLabel(this);
+        thirdRightPaneHeader->setObjectName("rightPaneHeader");
+        thirdRightPaneHeader->setText(tr("Enter your Details"));
+        thirdScreen->addWidget(thirdRightPaneHeader);
+
+        m_creatorName = new FormLineEditWidget(tr("Creator Name*"), "", this);
+        QLabel* creatorNameErrorLabel = m_creatorName->getErrorLabel();
+        creatorNameErrorLabel->setObjectName("createAGemFormErrorLabel");
+        QLineEdit* creatorNameLineEdit = m_creatorName->lineEdit();
+        creatorNameLineEdit->setPlaceholderText(tr("John Smith"));
+        thirdScreen->addWidget(m_creatorName);
+
+        m_repositoryURL = new FormLineEditWidget(tr("Repository URL*"), "", this);
+        QLabel* repositoryURLErrorLabel = m_repositoryURL->getErrorLabel();
+        repositoryURLErrorLabel->setObjectName("createAGemFormErrorLabel");
+        QLineEdit* repositoryURLLineEdit = m_repositoryURL->lineEdit();
+        repositoryURLLineEdit->setPlaceholderText(tr("https://github.com/Jane"));
+        thirdScreen->addWidget(m_repositoryURL);
+
+        return thirdScreenScrollArea;
+    }
+
+    bool CreateAGemScreen::ValidateGemDisplayName()
+    {
+        bool gemScreenNameIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            if (m_gemDisplayName->lineEdit()->text().isEmpty())
+            {
+                gemScreenNameIsValid = false;
+                m_gemDisplayName->setErrorLabelText(tr("A gem display name is required."));
+            }
+
+            m_gemDisplayName->setErrorLabelVisible(!gemScreenNameIsValid);
+        }
+        return gemScreenNameIsValid;
+    }
+
+    bool CreateAGemScreen::ValidateGemSystemName()
+    {
+        bool gemSystemNameIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            if (m_gemSystemName->lineEdit()->text().isEmpty())
+            {
+                gemSystemNameIsValid = false;
+                m_gemSystemName->setErrorLabelText(tr("A gem system name is required."));
+            }
+
+            m_gemSystemName->setErrorLabelVisible(!gemSystemNameIsValid);
+        }
+        return gemSystemNameIsValid;
+
+    }
+
+    bool CreateAGemScreen::ValidateLicenseName()
+    {
+        bool licenseNameIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            if (m_license->lineEdit()->text().isEmpty())
+            {
+                licenseNameIsValid = false;
+                m_license->setErrorLabelText(tr("License details are required."));
+            }
+
+            m_license->setErrorLabelVisible(!licenseNameIsValid);
+        }
+        return licenseNameIsValid;
+    }
+
+    bool CreateAGemScreen::ValidateCreatorName()
+    {
+        bool creatorNameIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            m_numNextClicks += 1;
+            if (m_numNextClicks > 1)
```

Comment:
This logic seems unnecessary - why are we tracking next clicks? can you not just always validate every field on the current tab widget page?

Reply:
Removed as I removed the connects to the various Validate functions in the constructor.